crates/uv/src/commands/self_update.rs

```diff
@@ -0,0 +1,75 @@
+use std::fmt::Write;
+
+use anyhow::Result;
+use axoupdater::{AxoUpdater, AxoupdateError};
+use owo_colors::OwoColorize;
+use uv_client::BetterReqwestError;
+
+use crate::commands::ExitStatus;
+use crate::printer::Printer;
+
+/// Attempt to update the `uv` binary.
+pub(crate) async fn self_update(printer: Printer) -> Result<ExitStatus> {
+    let mut updater = AxoUpdater::new_for("uv");
+
+    // Load the "install receipt" for the current binary. If the receipt is not found, then
+    // `uv` was likely installed via a package manager.
+    let Ok(receipt) = updater.load_receipt() else {
+        writeln!(
+            printer.stderr(),
+            "{}",
+            format_args!(
+                concat!(
+                    "{}{} Self-update is only available for `uv` binaries installed via the standalone installation scripts.",
+                    "\n",
+                    "\n",
+                    "If you installed `uv` with `pip`, `brew`, or another package manager, update `uv` with `pip install --upgrade`, `brew upgrade`, or similar."
+                ),
+                "warning".yellow().bold(),
+                ":".bold()
+            )
+        )?;
+        return Ok(ExitStatus::Error);
+    };
+
+    match receipt.run().await {
+        Ok(Some(result)) => {
+            writeln!(
+                printer.stderr(),
+                "{}",
+                format_args!(
+                    "{}{} Upgraded uv to {}! {}",
+                    "success".green().bold(),
+                    ":".bold(),
+                    format!("v{}", result.new_version).bold().white(),
+                    format!(
+                        "https://github.com/astral-sh/uv/releases/tag/{}",
+                        result.new_version
```

Comment:
The actual tag is available too, in case it differs from the version number.

```suggestion
                        result.new_version_tag
```